src/shared/components/AddPhoto.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  Divider,
+  IconButton,
+  ImageListItem,
+  Slide,
+  styled,
+  Typography,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import {TransitionProps} from '@mui/material/transitions';
+
+import {useAppDispatch, useAppSelector} from '../hooks/redux_hooks';
+import {clearPhotoState, getPhoto} from '../store/photoSlice';
+import UploadButton from './UploadButton/UploadButton';
+import {colors} from '../../styles/variables';
+import {IAlbumsProps} from '../interfaces/AlbumProps';
+import {AllPath} from '../constants/path';
+
+
+const MyImageListItem = styled(ImageListItem)(({selected}: { selected: boolean }) => ({
+  borderRadius: 8,
+  position: 'relative',
+  cursor: 'pointer',
+  backgroundColor: colors.light.colorIcon,
+  ['img']: {
+    transform: selected && 'scale(0.8)',
+  },
+}));
+
+const AddPhotoContainer = styled(Box)(() => ({
+
+}));
+
+const Transition = React.forwardRef(function transition(
+  props: TransitionProps & {
+    children: React.ReactElement;
+  },
+  ref: React.Ref<unknown>,
+) {
+  return <Slide direction="up" ref={ref} {...props} children={props.children}/>;
+});
+
+
+const AddPhoto = ({isOpen, handleOpen, setPhotos}: IAlbumsProps): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const {photo} = useAppSelector(state => state);
+  const [checkedPhoto, setCheckedPhoto] = useState<Record<number, boolean>>({});
+  const checkedPhotoLength = Object.entries(checkedPhoto).filter(item => item[1]);
+  const photos = Object.entries(checkedPhoto)
```

Comment:
Why `.entries()`? I think `.values()` would be more efficient in terms of code and won’t have to convert `items to `unknown`